### PR TITLE
Refactor Main Activity

### PR DESCRIPTION
**Pull request by linhpha**

### CODE REVIEW

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 1]
```diff
@@ -35,23 +34,22 @@
 import edu.grinnell.kdic.visualizer.VisualizeFragment;
 
 public class MainActivity extends AppCompatActivity {
-
+    //Set the tag for Main Activity class
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: You should remove this in-line comment. It's pretty obvious from the code.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 2]
```diff
@@ -65,151 +63,174 @@ public void onServiceConnected(ComponentName className, IBinder service) {
...
         }
-
         // Called when the connection with the service disconnects unexpectedly
         public void onServiceDisconnected(ComponentName className) {
-            Log.e(TAG, "onServiceDisconnected");
             boundToRadioService = false;
         }
     };
 
+    /**
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Do not comment inherited methods. **Only** comment methods you write

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 3]
```diff
@@ -65,151 +63,174 @@ public void onServiceConnected(ComponentName className, IBinder service) {
...
+     * @param savedInstanceState
+     */
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         // set the layout to use
         setContentView(R.layout.activity_main);
-
-        setupNavigation(); // setup the nav drawer and navigation functionality
+        // setup the nav drawer and navigation functionality
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: This method is obvious from the method name. You don't have to comment it.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 4]
```diff
@@ -65,151 +63,174 @@ public void onServiceConnected(ComponentName className, IBinder service) {
...
         setupPlaybackToolbar(); // setup the playback toolbar
 
         // update the schedule if connected to internet
         if (NetworkState.isOnline(this))
             updateSchedule();
-
     }
 
+    /**
+     *Bind the radio services after the activity is being displayed to the user
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Let's remove this comment

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 5]
```diff
@@ -65,151 +63,174 @@ public void onServiceConnected(ComponentName className, IBinder service) {
...
+     */
     @Override
     protected void onStart() {
         super.onStart();
-        Log.d(TAG, "onStart");
-
-        // bind to the radio service
+        // Start an intent to bind to the radio service
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: We should remove this as well

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 6]
```diff
@@ -65,151 +63,174 @@ public void onServiceConnected(ComponentName className, IBinder service) {
...
         startService(intent);
         bindService(intent, mConnection, BIND_AUTO_CREATE);
-
-        if (backStack.peek() != R.id.visualizer)
+        if (mBackStack.peek() != R.id.visualizer)
             updateShowNamePlaybackToolbar();
     }
 
+    /**
+     *Set up icon for the pause and play button
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Remove this comment

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 7]
```diff
@@ -65,151 +63,174 @@ public void onServiceConnected(ComponentName className, IBinder service) {
...
-            playPauseButton.setImageResource(R.drawable.ic_pause_white_24dp);
+            mPlayPauseButton.setImageResource(R.drawable.ic_pause_white_24dp);
         } else {
-            playPauseButton.setImageResource(R.drawable.ic_play_arrow_white_24dp);
+            mPlayPauseButton.setImageResource(R.drawable.ic_play_arrow_white_24dp);
         }
     }
 
+    /**
+     * Stop the activity
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Remove this as well

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 8]
```diff
@@ -65,151 +63,174 @@ public void onServiceConnected(ComponentName className, IBinder service) {
...
         unbindService(mConnection);
-
         boundToRadioService = false;
 
     }
 
+    /**
+     * Set up the playback bar on the bottom of the activity
+     */
     private void setupPlaybackToolbar() {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: I like that you comment this method. This method is supposed to set the Bottom navigation. let's keep it at exactly that. It's very long and we need to break it into smaller chunks. We can separate all of the onClick listeners into a `setButtomNavigationOnClickListeners()` and that'll cut it down.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 9]
```diff
@@ -222,85 +243,93 @@ private void setupFragments(Bundle savedInstanceState) {
...
-
-            visualizeFragment = new VisualizeFragment();
+            mVisualizeFragment = new VisualizeFragment();
         }
     }
 
+    /**
+     * Set up navigation drawer bar
+     */
     private void setupNavigation() {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: This method is too long.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 10]
```diff
@@ -361,25 +394,30 @@ public void onAnimationRepeat(Animation animation) {
         }
     }
 
+    /**
+     * Create visualizer fragment when the show is playing
+     */
     public void showVisualizeFragment() {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Let's shorten this method too

**Review by LarryAsante — CHANGES_REQUESTED** [review]
Nice work on this, Linh. There's some more work to be done on this:
* Let's break long methods to smaller methods
* Use static imports where necessary
* Remove in-line comments in which the code is obvious

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java (hunk 3 above)
Comment (linhpha) [reply app/src/main/java/edu/grinnell/kdic/MainActivity.java]: @LarryAsante I commited a new change in this pull request. Can you take a look at it? Thank you!

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 11]
```diff
@@ -302,58 +295,64 @@ public void onClick(View v) {
...
+                    //Generate favorite response
+                    ft.replace(R.id.fragment,
+                            mFavoritesFragment == null ? new FavoritesFragment() : mFavoritesFragment,
+                            FavoritesFragment.TAG)
+                            .addToBackStack(null)
+                            .commit();
+                    break;
+                case R.id.blog:
+                    //Generate blog fragment in response to the blog button
+                    ft.replace(R.id.fragment, new BlogWebViewFragment(), BlogWebViewFragment.TAG)
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Use a static import for the `TAG` here to shorten the code.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 12]
```diff
@@ -302,58 +295,64 @@ public void onClick(View v) {
...
+                case R.id.favorites:
+                    //Generate favorite response
+                    ft.replace(R.id.fragment,
+                            mFavoritesFragment == null ? new FavoritesFragment() : mFavoritesFragment,
+                            FavoritesFragment.TAG)
+                            .addToBackStack(null)
+                            .commit();
+                    break;
+                case R.id.blog:
+                    //Generate blog fragment in response to the blog button
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Remove this inline comment. I don't think it is necessary

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 13]
```diff
@@ -154,23 +143,11 @@ private void setupPlaybackToolbar() {
...
-                    if (mBackStack.peek() != R.id.visualizer) {
-                        showVisualizeFragment();
-                        mBackStack.add(R.id.visualizer);
-                    } else {
-                        hideVisualizeFragment();
-                        mBackStack.pop();
-                    }
-                    updateNavigationView();
-                }
+                setBottomNavigationOnClickListeners(v);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: I think the right method name for this, will be `toggleVisualizeFragmentOnClick(...)`

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 14]
```diff
@@ -182,6 +159,22 @@ public void onClick(View v) {
     }
 
     /**
+     * Set response when clicking in the bottom navigation
+     */
+    public void setBottomNavigationOnClickListeners(View v) {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: This doesn't really define what the function does. This toggles the Visualization Fragment so it should be named `toggleVisualizationFragment(..)` or `toggleVisualizationFragmentOnClick(..)`

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 15]
```diff
@@ -182,6 +159,22 @@ public void onClick(View v) {
     }
 
     /**
+     * Set response when clicking in the bottom navigation
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: This isn't too descriptive. It should tell us what happens on clicking the BottomBar (i.e Shows/hide the visualization fragment (replace this with : `{@link [name of the visualization fragment]` in the Javadoc of this method) on click)

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java (hunk 14 above)
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Are you using this method elsewhere? if no, please make this `private`

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 16]
```diff
@@ -302,58 +295,64 @@ public void onClick(View v) {
...
+     */
+    public boolean setOnClickNavigationItem(MenuItem menuItem) {
+        if (mBackStack.peek() != menuItem.getItemId()) {
+            if (mBackStack.peek() == R.id.visualizer) {
+                hideVisualizeFragment();
+                mBackStack.pop();
+            }
+            mBackStack.push(menuItem.getItemId());
+
+            //Create fragment transaction to handle response to call from the navigation drawer
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Remove the inline comments here. I don't think they are necessary.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 17]
```diff
@@ -302,58 +295,64 @@ public void onClick(View v) {
...
+                    ft.replace(R.id.fragment, new AboutFragment(), AboutFragment.TAG)
+                            .addToBackStack(null)
+                            .commit();
+                    break;
+                default:
+                    break;
+            }
+        }
+        // close the drawer after something is clicked
+        mDrawerLayout.closeDrawer(GravityCompat.START);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Use static import here

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 18]
```diff
@@ -394,12 +393,7 @@ public void onAnimationRepeat(Animation animation) {
             mPlayPauseButton.startAnimation(animation);
 
             // move the info onto the screen
-            AlphaAnimation alphaAnimation = new AlphaAnimation(0f, 1f);
-            alphaAnimation.setDuration(200);
-            alphaAnimation.setStartOffset(100);
-            alphaAnimation.setFillAfter(true);
-            alphaAnimation.setInterpolator(new AccelerateInterpolator());
-            findViewById(R.id.ll_show_info).startAnimation(alphaAnimation);
+            moveInfo(0f, 1f);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: You should extract these numbers as constants and give them descriptive names

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 19]
```diff
@@ -422,24 +416,39 @@ public void showVisualizeFragment() {
...
         }
     }
 
     /**
+     * Move the play button to the middle
+     */
+
+    public void movePlayButton(float shiftAmnt) {
+        TranslateAnimation animation = new TranslateAnimation(shiftAmnt, 0, 0, 0);
+        animation.setDuration(200);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: You should extract this number as a constant and give it a descriptive name. eg. `private final int MS_ANIMATION_DURATION = 200;`

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 20]
```diff
@@ -422,24 +416,39 @@ public void showVisualizeFragment() {
...
+
         }
     }
 
     /**
+     * Move the play button to the middle
+     */
+
+    public void movePlayButton(float shiftAmnt) {
+        TranslateAnimation animation = new TranslateAnimation(shiftAmnt, 0, 0, 0);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: What are these 0's here? I don't know what they are and forcing your client to always read documentation to understand random variables in a function isn't good for readability. Extract them as constants and give them proper naming. You might find it helpful to check the documentation of the constructor to see what those numbers are so you can name the constant properly

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 21]
```diff
@@ -422,24 +416,39 @@ public void showVisualizeFragment() {
...
+        animation.setFillAfter(true);
+
+    }
+
+    /**
+     * Create animation controls the alpha level
+     */
+    public void moveInfo(float fromAlpha, float toAlpha) {
+        AlphaAnimation alphaAnimation = new AlphaAnimation(fromAlpha, toAlpha);
+        alphaAnimation.setDuration(300);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Extract number as constant

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 22]
```diff
@@ -422,24 +416,39 @@ public void showVisualizeFragment() {
...
+        alphaAnimation.setStartOffset(100);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Same here. You can extract this as: `private final int MS_ANIMATION_START_OFFSET = 100;` I used the `MS` prefix here to indicate that it is in milliseconds. If it not, change it accordingly.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 23]
```diff
@@ -422,24 +416,39 @@ public void showVisualizeFragment() {
...
-
+            movePlayButton(shiftAmnt);
             // move the info off the screen
-            AlphaAnimation alphaAnimation = new AlphaAnimation(1f, 0f);
-            alphaAnimation.setDuration(300);
-            alphaAnimation.setStartOffset(100);
-            alphaAnimation.setFillAfter(true);
-            alphaAnimation.setInterpolator(new AccelerateInterpolator());
-            findViewById(R.id.ll_show_info).startAnimation(alphaAnimation);
+            moveInfo(1f, 0f);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Extract method parameters as constants with proper names. They seem like magic numbers here. (i.e Its hard to infer what they represent by just reading this line of code)

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 24]
```diff
@@ -302,58 +295,64 @@ public void onClick(View v) {
...
+            }
+        }
+        // close the drawer after something is clicked
+        mDrawerLayout.closeDrawer(GravityCompat.START);
+        return true;
+    }
+    /**
      * Highlight selected item
      */
     public void updateNavigationView() {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Should this method be public? If you're using it in just this class, you **should** make it `private`

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 25]
```diff
@@ -422,24 +416,39 @@ public void showVisualizeFragment() {
...
+            moveInfo(1f, 0f);
+
         }
     }
 
     /**
+     * Move the play button to the middle
+     */
+
+    public void movePlayButton(float shiftAmnt) {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Also check the access modifier. Should it be `public` or `private`?

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 26]
```diff
@@ -422,24 +416,39 @@ public void showVisualizeFragment() {
...
+        animation.setDuration(200);
+        animation.setInterpolator(new AccelerateDecelerateInterpolator());
+        animation.setFillAfter(true);
+
+    }
+
+    /**
+     * Create animation controls the alpha level
+     */
+    public void moveInfo(float fromAlpha, float toAlpha) {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Check access modifier

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 27]
```diff
@@ -512,9 +517,7 @@ private void updateSchedule() {
         getSchedule.execute();
     }
 
-    /**
-     * Return to previous activity on the stack,
-     */
+
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: This line is unnecessary. Remove it. You should format your code before pushing. To use our agreed code style, please see this (https://github.com/square/java-code-styles), install, set-up (select SquareAndroid in the preferred style) and format all of the classes you modify after you set it up

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 28]
```diff
@@ -34,7 +34,6 @@
 import edu.grinnell.kdic.visualizer.VisualizeFragment;
 
 public class MainActivity extends AppCompatActivity {
-    //Set the tag for Main Activity class
     public static final String TAG = MainActivity.class.getSimpleName();
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Should this tag be public? if you're not using it through out the app, it should stay private

**Review by LarryAsante — CHANGES_REQUESTED** [review]
Nice work @linhpha. I left a few comments. Please check them and make the necessary corrections/changes.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 29]
```diff
@@ -209,7 +201,7 @@ public void onClickPauseButton(View v) {
                             rotate = new RotateAnimation(0, 360, shiftX, shiftY);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Whoops! Looks like you missed this.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 30]
```diff
@@ -234,35 +226,28 @@ private void setupFragments(Bundle savedInstanceState) {
...
-
             // Add the fragment to the 'fragment_container' FrameLayout
             getSupportFragmentManager().beginTransaction()
                     .add(R.id.fragment, mScheduleFragment, ScheduleFragment.TAG)
                     .commit();
             mBackStack.add(R.id.schedule);
-
-
             SharedPreferences sharedPreferences = getSharedPreferences(Constants.SHARED_PREFS, 0);
             if (sharedPreferences.getBoolean(Constants.FIRST_RUN, true)) {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Use a static import here

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java (hunk 29 above)
Comment (linhpha) [reply app/src/main/java/edu/grinnell/kdic/MainActivity.java]: @LarryAsante  I pushed again

**Comment by LarryAsante** [comment]
@linhpha There's a conflict but hold on a second. @nikagarwal98 made changes to main. I'll check and merge that so you can pull and make changes to `RadioService.java`

**Pull request closed without merging**